.github/workflows/build_and_test.yml

```diff
@@ -32,7 +32,7 @@ jobs:
     - uses: actions/checkout@v4
       with:
         fetch-depth: '0'
-        submodules: true
+        submodules: recursive
```

Comment:
Technically this wouldn't be needed anymore as we no longer have a submodule that contains a submodule.

`quantinuum-sphinx` is now installable from the github repo rather than a submodule of `pytket-docs-theming`.

Reply:
I don't see any particular downside to keeping it recursive until someone adds something huge as a submodule, which would likely not be a good thing to do anyway; but I don't feel very strongly either way